account_product_fiscal_classification/models/product_template.py

```diff
@@ -75,15 +77,58 @@ def _update_vals_fiscal_classification(self, vals):
                 }
             )
         elif vals.get("supplier_taxes_id") or vals.get("taxes_id"):
-            raise ValidationError(
-                _(
-                    "You can not create or write products with"
-                    " 'Customer Taxes' or 'Supplier Taxes'\n."
-                    "Please, use instead the 'Fiscal Classification' field."
-                )
-            )
+            self._find_or_create_classification(vals)
         return vals
 
     @api.constrains("categ_id", "fiscal_classification_id")
     def _check_rules_fiscal_classification(self):
         self.env["account.product.fiscal.rule"].check_product_templates_integrity(self)
+
+    def _find_or_create_classification(self, vals):
+        """Find the correct Fiscal classification,
+        depending of the taxes, or create a new one, if no one are found."""
+        # search for matching classication
+        domain = []
+        purchase_tax_ids = vals.get("supplier_taxes_id")
+        sale_tax_ids = vals.get("taxes_id")
+        for elm in ("supplier_taxes_id", "taxes_id"):
+            if elm in vals:
+                del vals[elm]
+        taxe_ids = []
+        if sale_tax_ids:
+            if isinstance(sale_tax_ids, int):
```

Comment:
in which situation sale_tax_ids is an int ? I don't see such test in Odoo core modules (or in OCA modules). could you elaborate ?